inc/Store/DataSource/DataSourceConfigManager.php

```
@@ -0,0 +1,248 @@
+use WP_Error;
+
+class DataSourceConfigManager {
+	public const CONFIG_SOURCE_CODE = 'code';
+	public const CONFIG_SOURCE_STORAGE = 'storage';
+	public const CONFIG_SOURCE_CONSTANT = 'constant';
+	public const MUTABLE_CONFIG_SOURCES = [ self::CONFIG_SOURCE_STORAGE ];
+
+	private static function get_all_from_storage(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_STORAGE ] );
+			},
+			DataSourceCrud::get_configs()
+		);
+	}
+
+	private static function get_all_from_constant(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_CONSTANT ] );
+			},
+			ConstantConfigStore::get_configs()
+		);
+	}
+
+	private static function get_all_from_code(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_CODE ] );
+			},
+			ConfigStore::get_data_sources_as_array()
+		);
+	}
+
+	/**
+	 * Quick and dirty de-duplication of data sources. If the data source does
+	 * not have a UUID (because it is registered in code), we generate an
+	 * identifier based on the display name and service name.
+	 */
+	private static function de_duplicate_configs( array $configs ): array {
+		return array_values( array_reduce(
+			$configs,
+			function ( array $acc, array $item ) {
+				$identifier = $item['uuid'] ?? md5(
+					sprintf( '%s_%s', $item['service_config']['display_name'], $item['service'] )
+				);
+				$acc[ $identifier ] = $item;
+				return $acc;
+			},
+			[]
+		) );
+	}
+
+	/**
+	 * Get all data sources from all origins with optional filters.
+	 * 
+	 * Supported filters:
+	 * - service: Filter by service name (e.g. 'airtable', 'google-sheets', 'shopify')
+	 * - enable_blocks: Filter by blocks enabled status (false matches with null/false and true matches with true)
+	 * 
+	 * Any unsupported filter keys will be ignored.
```

I'd prefer to return an error

---

61d87d3

---

Looks like the comment needs to be updated as well, with the latest changes?